src/utils/wallet.js

```diff
@@ -256,13 +397,45 @@ class Wallet {
         }
     }
 
-    async createNewAccount(accountId, fundingKey, fundingContract) {
+    /********************************
+    Check account was created before saving 10 retries, 1s each
+    ********************************/
+    async checkAccountAndSave(accountId, keyPair) {
+        return new Promise((resolve, reject) => {
+            let attempts = 0
+            const retries = 10
+            const check = async () => {
+                attempts++
+                console.log(`attempt: ${attempts}, checking for account: ${accountId}`)
+                if (attempts >= retries) {
+                    reject('account cannot be found')
+                    return
+                }
+                try {
+                    await this.getAccount(accountId).state()
+                    resolve(await this.saveAndSelectAccount(accountId, keyPair));
+                } catch (e) {
+                    setTimeout(check, 1000)
+                }
+            }
+            check()
+        })
+    }
+
+    async createNewAccount(accountId, fundingContract, fundingKey) {
```

Comment:
not sure what is the reason for arbitrary changes to handling linkdrop (like reordering arguments). There is already plenty of other stuff in this PR :)

Reply:
the order of the arguments didn't match the order in the route params, this led to a lot of confusion so we switched them for consistentcy.